File: Daily challenge/Trapping Rain Water.cpp
class Solution 
{
public:
    int trap(vector<int>& height) 
    {
        int n = height.size();
        int left = height[0], right = height[n - 1];
        int res = 0;
        for(int i = 0, j = n - 1; i < j;)
        {
            if(left <= right)
            {
                res += left - height[i];
                i++;
                left = max(left, height[i]);
            }
            else
            {
                res += right - height[j];
                j--;
                right = max(right, height[j]);
            }
        }
        return res;
    }
};

File: Daily challenge/Trapping Rain Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var trap = function(height) 
{
    let n = height.length;
    let left = height[0];
    let right = height[n - 1];
    let res = 0;
    let i = 0;
    let j = n - 1;
    while (i < j) 
    {
        if (left <= right) 
        {
            res += left - height[i];
            i++;
            left = Math.max(left, height[i]);
        } 
        else 
        {
            res += right - height[j];
            j--;
            right = Math.max(right, height[j]);
        }
    }
    return res;    
};